### PR TITLE
add support for discovery plugin

### PR DESCRIPTION
**Pull request by brandonweeks**
Perhaps installing the discovery image should be optional? Here's an initial PR without it.

### CODE REVIEW

**Comment by brandonweeks** [comment]
I went ahead and added an $install_images param to match the discovery plugin code from theforeman/puppet-foreman. However I'm unsure if using the smart_proxy_discovery_image package added in theforeman/foreman-packaging@5eb2ef27e17b6101f00205d296e1064ded49c8f8 is the best way going forward or if this should be using the download method from puppet-foreman/manifests/plugin/discovery.pp.

@lzap thoughts?

**Comment by domcleal** [comment]
I don't think users are meant to install the smart_proxy_discovery_image plugin directly, but instead should be downloading an image (or building their own).

I think the $install_images parameter should be here too and we should have the same implementation as puppet-foreman.  It seems like it's needed either in both depending on whether the smart proxy plugin is in use or not.

**Comment by brandonweeks** [comment]
@domcleal thank you for the explanation on IRC. 

I've updated the PR to mirror what puppet-foreman is doing with slightly adjusted tests.

File: manifests/plugin/discovery.pp [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# = Foreman Proxy Discovery plugin
+#
+# This class installs discovery plugin and images on smart proxys
+#
+# === Parameters:
+#
+# $install_images::  should the installer download and setup discovery images
```
Comment (ekohl) [inline manifests/plugin/discovery.pp]: I'd prefer leaving out the installer and have a more generic description like "should discovery images be downloaded and setup?"

**Comment by ekohl** [comment]
Code wise it looks good.

**Comment by brandonweeks** [comment]
@ekohl reworded

**Comment by ekohl** [comment]
:+1: on green tests.

**Comment by mmoll** [comment]
the test errors are probably related to the new rspec-puppet-facts (or more precise: facterdb) release...

**Comment by domcleal** [comment]
I've filed a bug over on facterdb as it appears the contents of some facts has changed (but not in GH!): https://github.com/camptocamp/facterdb/issues/14

**Comment by lzap** [comment]
LGTM

**Comment by domcleal** [comment]
> I've filed a bug over on facterdb as it appears the contents of some facts has changed (but not in GH!): camptocamp/facterdb#14

They changed, but I think the onus is on us to set the facts correctly in the specs.  We can't assume that r-p-f/facterdb provides either non-AIO or AIO style facts.

**Comment by mmoll** [comment]
I'm in favour of merging this and have a look into fixing the tests later (or pinning rspec-puppet-facts for now as a temporary bandaid).

**Comment by ekohl** [comment]
@mmoll I agree. I'm sure master is already broken by puppetdb/rspec-puppet-facts so it's not getting any worse.

**Comment by mmoll** [comment]
merged, thanks @brandonweeks!

**Pull request merged**